rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1660,23 +1659,28 @@ public NodeSet getExactlyNodes(int nb, SelectionScript selectionScript) {
      * @return RMInitialState containing nodes and nodeSources of the RMCore.
      */
     public RMInitialState getRMInitialState() {
-        Collection<RMNode> nodes = this.allNodes.values();
-        ArrayList<RMNodeEvent> nodesList = new ArrayList<>(nodes.size());
 
-        for (RMNode rmnode : nodes) {
-            nodesList.add(rmnode.createNodeEvent());
+        Map<String, RMNodeEvent> nodeEvents = new HashMap<>();
+        for (RMNode rmnode : this.allNodes.values()) {
+            final RMNodeEvent nodeEvent = rmnode.createNodeEvent();
+            nodeEvents.put(nodeEvent.getNodeUrl(), nodeEvent);
         }
 
-        Collection<NodeSource> nodeSources = this.nodeSources.values();
-        ArrayList<RMNodeSourceEvent> nodeSourcesList = new ArrayList<>(nodeSources.size());
-        for (NodeSource s : nodeSources) {
-            nodeSourcesList.add(new RMNodeSourceEvent(s.getName(), s.getDescription(), s.getAdministrator().getName()));
-            for (RMDeployingNode pn : s.getDeployingAndLostNodes()) {
-                nodesList.add(pn.createNodeEvent());
+        Map<String, RMNodeSourceEvent> nodeSources = new HashMap<>();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "nodeSources" which hides the field declared at line 221. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)